### ISSUE TITLE
Appended (2) to received shares

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/server#1588]
* userA shares a mp3 file to userB
* userB sees this file
* userB uses the Android app (current version in the appstore) to open that mp3 file
* userB switches from portrait mode to landscape while it is playing the mp3
* the file gets an (2) appended (and all other received shares from userA too)

cc @AndyScherzinger @tobiasKaminsky What is done, when switching the screen mode?

This is on 10.0.0 - nothing in the logs.

@nickvergessen @blizzz because you also digged into this problem.

@schiessle @icewind1991 I talked with you about this too.

**Comment by MorrisJobke** [comment]
Seems to be related to https://github.com/owncloud/core/issues/25718

**Comment by MorrisJobke** [comment]
Okay. I tried to reproduce this and I failed. Next step: I will debug this via proxy

**Comment by MorrisJobke** [comment]
I replayed all requests that are logged and found the cause:

```
$ curl -H 'Authorization:Basic TOKEN' -H 'User-Agent:Mozilla/5.0 (Android) ownCloud-android/1.3.1' -H 'host:cloud.morrisjobke.de' 'https://cloud.morrisjobke.de/index.php/avatar/Username/384'
{"data":{"displayname":"username"}}
``` 

As you see the username has a capital letter in the URL. If I change that to a lowercase version nothing bad happens, but if this is with the capitalized username it appends a `(2)`.

**Comment by AndyScherzinger** [comment]
@MorrisJobke withoput looking into the code the screen rotation shouldn't trigger anything. Only thing I can think of (which might then be a bug though) would be a second download of the file.

**Comment by MorrisJobke** [comment]
Steps to reproduce:

* setup user `abc` and add an avatar to the users profile
* share a file to user `abc`
* as user `abc` request the avatar like this:
```
curl -i -u abc:abc 'http://localhost/index.php/avatar/Abc/384'
```

If the avatar URL has a lowercase username the avatar is returned - with the above URL following is returned:
```json
{"data":{"displayname":"abc"}}
```
Then also the (2) is appended to every shared file.

@rullzer maybe you want to dig into that - otherwise I will spend some time debugging this ;)

**Comment by MorrisJobke** [comment]
The trace of the exception in https://github.com/nextcloud/server/blob/6740c627b87157a16c1cce554307dce807a92602/core/Controller/AvatarController.php#L146

```json
{"reqId":"lJOd4tUhQ\/q2Y95fjAcu","remoteAddr":"127.0.0.1","app":"no app in context","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\NotFoundException\",\"Message\":\"\\\/Abc\",\"Code\":0,\"Trace\":\"
#0 [internal function]: OC\\\\Files\\\\Node\\\\Root->get('\\\/Abc')
#1 lib\\\/private\\\/Files\\\/Node\\\/LazyRoot.php(65): call_user_func_array(Array, Array)
#2 lib\\\/private\\\/Files\\\/Node\\\/LazyRoot.php(142): OC\\\\Files\\\\Node\\\\LazyRoot->__call('get', Array)
#3 lib\\\/private\\\/AvatarManager.php(103): OC\\\\Files\\\\Node\\\\LazyRoot->get('\\\/Abc')
#4 core\\\/Controller\\\/AvatarController.php(132): OC\\\\AvatarManager->getAvatar('Abc')
#5 [internal function]: OC\\\\Core\\\\Controller\\\\AvatarController->getAvatar('Abc', 384)
#6 lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)
#7 lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Core\\\\Controller\\\\AvatarController), 'getAvatar')
#8 lib\\\/private\\\/AppFramework\\\/App.php(127): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Core\\\\Controller\\\\AvatarController), 'getAvatar')
#9 lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OC\\\\\\\\Core\\\\\\\\Control...', 'getAvatar', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)
#10 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)
#11 lib\\\/private\\\/Route\\\/Router.php(298): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)
#12 lib\\\/base.php(994): OC\\\\Route\\\\Router->match('\\\/avatar\\\/Abc\\\/384')
#13 index.php(40): OC::handleRequest()
#14 {main}\",\"File\":\"lib\\\/private\\\/Files\\\/Node\\\/Root.php\",\"Line\":189}","level":3,"time":"2016-09-30T12:51:36+00:00","method":"GET","url":"\/index.php\/avatar\/Abc\/384","user":"abc"}
```

**Comment by MorrisJobke** [comment]
Fix is in https://github.com/nextcloud/server/pull/1595

**Comment by MorrisJobke** [comment]
At least my issue is fixed with #1595 - @blizzz Could you give the patch a try - because you also reported this ;)

**Comment by blizzz** [comment]
> At least my issue is fixed with #1595 - @blizzz Could you give the patch a try - because you also reported this ;)

I applied the Nc10 version, let's see.
